### PR TITLE
Fix failures when performing binary operations on DataFrames with empty columns

### LINKED ISSUE #8434: [BUG] Binary operations on DataFrames and Series with empty columns fail
**Describe the bug**
Performing binary operations on empty DataFrames or Series throws an error.

**Steps/Code to reproduce bug**
For example:
```
import cudf

df = cudf.DataFrame(columns=["a", "b"])
df / 2
```
returns
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-13-a74c9cd24885> in <module>
----> 1 df / 2

~/anaconda3/envs/cudf_dev/lib/python3.8/site-packages/cudf/core/dataframe.py in __truediv__(self, other)
   2393 
   2394     def __truediv__(self, other):
-> 2395         return self._apply_op("__truediv__", other)
   2396 
   2397     def rtruediv(self, other, axis="columns", level=None, fill_value=None):

~/anaconda3/envs/cudf_dev/lib/python3.8/site-packages/cudf/core/dataframe.py in _apply_op(self, fn, other, fill_value)
   1539         ):
   1540             for col in self._data:
-> 1541                 result[col] = op(self[col], other)
   1542         else:
   1543             raise NotImplementedError(

~/anaconda3/envs/cudf_dev/lib/python3.8/site-packages/cudf/core/dataframe.py in op(lhs, rhs)
   1470         def op(lhs, rhs):
   1471             if fill_value is None:
-> 1472                 return getattr(lhs, fn)(rhs)
   1473             else:
   1474                 return getattr(lhs, fn)(rhs, fill_value)

~/anaconda3/envs/cudf_dev/lib/python3.8/site-packages/cudf/core/frame.py in __truediv__(self, other)
   3797             return self._binaryop(other, "div")
   3798         else:
-> 3799             return self._binaryop(other, "truediv")
   3800 
   3801     def __rtruediv__(self, other):

~/anaconda3/envs/cudf_dev/lib/python3.8/site-packages/cudf/core/series.py in _binaryop(self, other, fn, fill_value, reflect, can_reindex)
   1340             lhs = self
   1341 
-> 1342         return super()._binaryop(other, fn, fill_value, reflect, lhs)
   1343 
   1344     def add(self, other, fill_value=None, axis=0):

~/anaconda3/envs/cudf_dev/lib/python3.8/site-packages/cudf/core/frame.py in _binaryop(self, other, fn, fill_value, reflect, lhs, *args, **kwargs)
   3670             lhs = self
   3671 
-> 3672         rhs = self._normalize_binop_value(other)
   3673 
   3674         if fn == "truediv":

~/anaconda3/envs/cudf_dev/lib/python3.8/site-packages/cudf/core/frame.py in _normalize_binop_value(self, other)
   3727             return cudf.Scalar(other, dtype=self.dtype)
   3728         else:
-> 3729             return self._column.normalize_binop_value(other)
   3730 
   3731     def _bitwise_binop(self, other, op):

~/anaconda3/envs/cudf_dev/lib/python3.8/site-packages/cudf/core/column/string.py in normalize_binop_value(self, other)
   5348             return col
   5349         else:
-> 5350             raise TypeError(f"cannot broadcast {type(other)}")
   5351 
   5352     def default_na_value(self) -> ScalarLike:

TypeError: cannot broadcast <class 'int'>
```
and
```
df.a / 2
```
returns
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-ee23a02e5df2> in <module>
----> 1 df.a/2

~/anaconda3/envs/cudf_dev/lib/python3.8/site-packages/cudf/core/frame.py in __truediv__(self, other)
   3797             return self._binaryop(other, "div")
   3798         else:
-> 3799             return self._binaryop(other, "truediv")
   3800 
   3801     def __rtruediv__(self, other):

~/anaconda3/envs/cudf_dev/lib/python3.8/site-packages/cudf/core/series.py in _binaryop(self, other, fn, fill_value, reflect, can_reindex)
   1340             lhs = self
   1341 
-> 1342         return super()._binaryop(other, fn, fill_value, reflect, lhs)
   1343 
   1344     def add(self, other, fill_value=None, axis=0):

~/anaconda3/envs/cudf_dev/lib/python3.8/site-packages/cudf/core/frame.py in _binaryop(self, other, fn, fill_value, reflect, lhs, *args, **kwargs)
   3670             lhs = self
   3671 
-> 3672         rhs = self._normalize_binop_value(other)
   3673 
   3674         if fn == "truediv":

~/anaconda3/envs/cudf_dev/lib/python3.8/site-packages/cudf/core/frame.py in _normalize_binop_value(self, other)
   3727             return cudf.Scalar(other, dtype=self.dtype)
   3728         else:
-> 3729             return self._column.normalize_binop_value(other)
   3730 
   3731     def _bitwise_binop(self, other, op):

~/anaconda3/envs/cudf_dev/lib/python3.8/site-packages/cudf/core/column/string.py in normalize_binop_value(self, other)
   5348             return col
   5349         else:
-> 5350             raise TypeError(f"cannot broadcast {type(other)}")
   5351 
   5352     def default_na_value(self) -> ScalarLike:

TypeError: cannot broadcast <class 'int'>
```

**Expected behavior**
Performing the same operations in Pandas returns an empty Series/DataFrame. For example:
```
import pd

df = pd.DataFrame(columns=["a","b"])
df / 2
```
returns
```
Empty DataFrame
Columns: [a, b]
Index: []
``` 
and
```
df.a / 2
```
returns
```
Series([], Name: a, dtype: object)
```

### PR DESCRIPTION
**Pull request by ChrisJar**
Closes #8434

### CODE REVIEW

**Review by galipremsagar — CHANGES_REQUESTED** [review]

**Comment by beckernick** [comment]
@ChrisJar is this on target for 21.08 or should we push to 21.10?

**Comment by ChrisJar** [comment]
Yep! This is still on track I'm just finishing up some refactoring that should be done by this afternoon

File: python/cudf/cudf/core/series.py [hunk 1]
```diff
@@ -1326,6 +1326,16 @@ def _binaryop(
         if isinstance(other, cudf.DataFrame):
             return NotImplemented
 
+        # Ignore empty object columns when performing arithmetic operations
+        if (
+            self.dtype == "object"
+            and self.isnull().all()
```
[the remark this answers is not in the document]
Comment (ChrisJar) [reply python/cudf/cudf/core/series.py]: So the issue is that in Pandas performing an arithmetic binary operation on empty or an all null object columns returns the column unchanged and performing a comparison operation returns the appropriate boolean column. However, in cudf both of these actions currently throw. The behavior is already correct for other column types it just needs to be fixed for object columns

**Review by galipremsagar — APPROVED** [review]

**Review by galipremsagar — APPROVED** [review]

**Review by charlesbluca — APPROVED** [review]

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**